app/models/api/trainee_filter_params.rb

```diff
@@ -46,5 +39,9 @@ def check_academic_cycle
         )
       end
     end
+
+    def iso8601?(string)
+      string.match?(/\A\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(Z|(\+|\-)\d{2}:\d{2})?\z/)
+    end
```

Comment:
I'm not a big fan of complex regexes like this because they are pretty hard to parse at a glance and see what's going on and if the standard were to change (however unlikely) we'd have to update this. Could we use the built-in `iso8601` method to try to parse the string?

Reply:
Opened a [draft PR](https://github.com/DFE-Digital/register-trainee-teachers/pull/4289) to demo which cases won't be captured if we use the `DateTime.iso6801` as an example.